submissions/OlStaniev/OOP exercise/index.js

```diff
@@ -0,0 +1,44 @@
+class Inhabitant {
+   constructor(name, gender, legs, saying) {
+      this.name = name
+      this.gender = gender
+      this.legs = legs
+      this.saying = saying
+   }
+   prepareToPrint() {
+      return [this.name, this.gender, this.legs, this.saying].join('; ')
+   }
+}
+
+class Person extends Inhabitant {
+   constructor(name, gender, legs, saying, hands, friends) {
+      super(name, gender, legs, saying)
+      this.specie = 'human'
+      this.hands = hands
+      this.friends = friends || []
+   }
+   prepareToPrint() {
+      return this.specie + '; ' + super.prepareToPrint() + '; ' + [this.hands, this.friends.join(', ')].join('; ')
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(specie, name, gender, legs, saying, tail) {
+      super(name, gender, legs, saying)
+      this.specie = specie
```

Comment:
Now we have duplication of properties in class hierarchy that violates OOP/SOLID principles.
Shared properties must be owned by a base class and handled in one class.